src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -44,6 +45,12 @@ public TelemetryClient(TelemetryConfiguration configuration)
             }
 
             this.configuration = configuration;
+
+            // If someone created configuration from scratch and forgot to initialize the channel then use the default.
+            if (this.configuration.TelemetryChannel == null)
+            {
+                throw new ArgumentException("The specified configuration does not have a telemetry channel.", "configuration");
```

Comment:
P.S. @SergeyKanzhelev I have preserved the behavior of the scenario's we discussed, this throw will not happen for any existing customer but it does help improve for future if someone explicitly tries the following silly code:
```
var conf = new TelemetryConfigration();
conf.TelemetryChannel = null;
var client = new TelemetryClient(conf);
```